flexmeasures/data/models/data_sources.py

```diff
@@ -144,3 +166,21 @@ def to_dict(self) -> dict:
             type=self.type if self.type in ("forecaster", "scheduler") else "other",
             description=self.description,
         )
+
+    def get_attribute(self, attribute: str, default: Any = None) -> Any:
+        """Looks for the attribute on the DataSource.
+        If not found, returns the default.
+        """
+        if hasattr(self, attribute):
```

Comment:
> Likewise for the custom parameters. That is, we shouldn't use name as one of the key names in the attributes column right?

Good point. As the parameters are wrap under the `config` key, there are not reserved keys. Nonetheless, we can't add config as a class attribute.

Reply:
With my last comment I actually meant that one of the DataSource column names (`id`, `name`, `type`, `user_id`, model`, `version` and `attributes`) shouldn't be used as a key name in the attributes column, because it won't be found by this logic.

Reply:
True! I'm removing the option to access the class attributes.